yash-datalake/nested/azureFunction.json

```diff
@@ -1,406 +1,359 @@
-				"http": {
-					"routePrefix": ""
-				}
-			},
-			"dependsOn": [
-				"[resourceId('Microsoft.Web/sites', variables('functionAppName'))]",
-				"[resourceId('Microsoft.Web/serverfarms', variables('hostingPlanName'))]",
-				"[concat('Microsoft.Web/sites/',variables('functionAppName'),'/Extensions/MSDeploy')]"
-			]
-		}
-	]
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "location": {
+      "type": "string",
+      "metadata": {
+        "description": "Deployment location"
+      }
+    },
+    "KibanaESDeployedVMIP": {
+      "type": "string",
+      "metadata": {
+        "description": "Public IP of VM on which kibana install."
+      }
+    },
+    "web_app_id": {
+      "type": "string",
+      "metadata": {
+        "description": "The ID of the service principal web/api"
+      }
+    },
+    "web_app_secret": {
+      "type": "string",
+      "metadata": {
+        "description": "The ID of the service principal web/api"
+      }
+    },
+    "appName": {
+      "type": "string",
+      "metadata": {
+        "description": "Web application name"
+      }
+    },
+    "datafactory_name": {
+      "type": "string",
+      "metadata": {
+        "description": "The ID of the service principal"
+      }
+    },
+    "storageAccountName": {
+      "type": "string",
+      "metadata": {
+        "description": "storage account name"
+      }
+    },
+    "storageAccountConnectionString": {
+      "type": "string",
+      "metadata": {
+        "description": "storage account connection string"
+      }
+    },
+    "eventhub_name": {
+      "type": "string",
+      "metadata": {
+        "description": "Event hub name"
+      }
+    },
+    "eventhub_namespace": {
+      "type": "string",
+      "metadata": {
+        "description": "eventhub namespace name"
+      }
+    },
+    "sas_key": {
+      "type": "string",
+      "metadata": {
+        "description": " event hub SharedAccessPolicyPrimaryKey"
+      }
+    },
+    "adla_account_name": {
+      "type": "string",
+      "metadata": {
+        "description": "Azure datalake analytics name"
+      }
+    },
+    "DataCleaningStreamAnalyticsJob": {
+      "type": "String",
+      "metadata": {
+        "description": " Azure stream analytics job name 1"
+      }
+    },
+    "DataAggregatorStreamAnalyticsJob1": {
+      "type": "String",
+      "metadata": {
+        "description": " Azure stream analytics job name 1"
+      }
+    },
+    "DataAggregatorStreamAnalyticsJob2": {
+      "type": "String",
+      "metadata": {
+        "description": " Azure stream analytics job name 2"
+      }
+    },
+    "DataAggregatorStreamAnalyticsJob3": {
+      "type": "String",
+      "metadata": {
+        "description": " Azure stream analytics job name 3"
+      }
+    }
+  },
+  "variables": {
+    "_artifactsLocation": "https://github.com/ajos1993/yash-datalake/raw/master/",

```

Comment:
See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#deployment-artifacts-nested-templates-scripts) for how to handle artifacts